src/future/cache.rs

```diff
@@ -3062,6 +3062,44 @@ mod tests {
         expiry_counters.verify();
     }
 
+    // https://github.com/moka-rs/moka/issues/345
+    #[tokio::test]
+    async fn test_race_between_updating_entry_and_processing_its_write_op() {
+        let cache = Cache::builder()
+            .max_capacity(2)
+            .time_to_idle(Duration::from_secs(1))
+            .build();
+        let (clock, mock) = Clock::mock();
+        cache.set_expiration_clock(Some(clock)).await;
+
+        cache.insert("a", "alice").await;
+        cache.insert("b", "bob").await;
+        cache.insert("c", "cathy").await; // c1
+        mock.increment(Duration::from_secs(2));
+
+        // The following `insert` will do the followings:
+        // 1. Replaces current "c" (c1) in the concurrent hash table (cht).
+        // 2. Runs the pending tasks implicitly.
+        //    (1) c1 will be evicted by size constraint.
+        //    (2) "a" will be admitted.
+        //    (3) "b" will be admitted.
```

Comment:
```suggestion
        //    (1) "a" will be admitted.
        //    (2) "b" will be admitted.
        //    (3) c1 will be evicted by size constraint.
```

Reply:
Applied the change:
https://github.com/moka-rs/moka/pull/348/commits/a28970ca7f1ab68a7adaa802b3739efb1e3f5212#diff-338f772312880199137f8bbb508f15dd7f83457565cc983b85043cddd71b2f24